app/src/main/java/com/andrew/tasky/presentation/agenda_item_detail/AgendaItemDetailFragment.kt

```diff
@@ -334,18 +350,44 @@ class AgendaItemDetailFragment: Fragment(R.layout.fragment_agenda_item_detail) {
         popupMenu.show()
     }
 
+    private fun getAgendaItemCurrentInfo(): AgendaItem {
+        return AgendaItem(
+            agendaItemType,
+            viewModel.isDone.value,
+            viewModel.title.value,
+            viewModel.description.value,
+            LocalDateTime.of(viewModel.selectedDate.value, viewModel.selectedTime.value),
+            reminderTime = viewModel.selectedReminderTime.value
+        )
```

Comment:
That's also a function, you could completely move to the ViewModel, since it only depends on UI state which also sits there.